extensions/vertx-web-common/deployment/src/main/java/io/quarkus/vertx/web/common/deployment/VertxWebCommonProcessor.java

```diff
@@ -0,0 +1,83 @@
+import io.quarkus.vertx.web.common.runtime.RouterProducer;
+import io.quarkus.vertx.web.common.runtime.VertxWebCommonRecorder;
+import io.quarkus.vertx.web.common.runtime.cors.CORSRecorder;
+import io.vertx.ext.web.Router;
+
+class VertxWebCommonProcessor {
+
+    HttpConfiguration httpConfiguration;
+
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    FilterBuildItem cors(CORSRecorder recorder, HttpConfiguration configuration) {
+        return new FilterBuildItem(recorder.corsHandler(configuration));
+    }
+
+    @BuildStep
+    AdditionalBeanBuildItem additionalBeans() {
+        return AdditionalBeanBuildItem.unremovableOf(RouterProducer.class);
+    }
+
+    @BuildStep(onlyIf = IsNormal.class)
+    @Record(value = ExecutionTime.RUNTIME_INIT, optional = true)
+    public KubernetesPortBuildItem kubernetes(HttpConfiguration config, BuildProducer<KubernetesPortBuildItem> portProducer,
+            VertxWebCommonRecorder recorder) {
+        int port = ConfigProvider.getConfig().getOptionalValue("quarkus.http.port", Integer.class).orElse(8080);
+        recorder.warnIfPortChanged(config, port);
+        return new KubernetesPortBuildItem(config.port, "http");
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    InternalVertxWebRouterBuildItem initializeRouter(VertxWebCommonRecorder recorder, BeanContainerBuildItem beanContainer,
+            LaunchModeBuildItem launchMode,
+            ShutdownContextBuildItem shutdown,
+            InternalVertxBuildItem vertx,
+            Optional<RequireVirtualHttpBuildItem> requireVirtual) throws IOException {
+
+        boolean startVirtual = requireVirtual.isPresent() || httpConfiguration.virtual;
+        // start http socket in dev/test mode even if virtual http is required
+        boolean startSocket = !startVirtual || launchMode.getLaunchMode() != LaunchMode.NORMAL;
+        RuntimeValue<Router> router = recorder.initializeRouter(vertx.getVertx(), beanContainer.getValue(),
+                httpConfiguration, launchMode.getLaunchMode(), startVirtual, startSocket);
+
+        return new InternalVertxWebRouterBuildItem(router);
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    ServiceStartBuildItem finalizeRouter(VertxWebCommonRecorder recorder, BeanContainerBuildItem beanContainer,
+            LaunchModeBuildItem launchMode, ShutdownContextBuildItem shutdown, Optional<DefaultRouteBuildItem> defaultRoute,
+            List<FilterBuildItem> filters, InternalVertxWebRouterBuildItem router,
+            List<AdditionalRoutesInstalledBuildItem> additionalRoutesInstalled) {
+        recorder.finalizeRouter(beanContainer.getValue(), defaultRoute.map(DefaultRouteBuildItem::getHandler).orElse(null),
+                filters.stream().map(FilterBuildItem::getHandler).collect(Collectors.toList()),
+                launchMode.getLaunchMode(), shutdown);
+
+        return new ServiceStartBuildItem("vertx-web");
```

Comment:
I would change this name to `vertx-http`.

Reply:
Done.